tools/toolchains/mbed_toolchain.py

```diff
@@ -895,62 +895,52 @@ def add_regions(self):
         """Add regions to the build profile, if there are any.
         """
```

Comment:
```suggestion
         """
         if not getattr(self.target, "bootloader_supported", False):
            return
```

Reply:
Commit msg (explanation): 

```
om/JammuKekkonen/mbed-os into JammuKekkonen-remove_hiding_config_errors)
Branch: JammuKekkonen-remove_hiding_config_errors
Follows: mbed-os-5.4.0-rc1
Precedes: 

    tools: regions only if bootloader is supported
    
    We removed catching and passing, we want to know for any misconfiguration if bootloader
    is supported. Regions should check if bootloader is supported. In case not, just return.
    Otherwise we catch any error.
    
    This should help us to uncover missing regions or other config error (in case bootloader
    is enabled via bootloader_supported set to true).
```

Reply:
I was not able to push to your branch .  I tested this for `WIZWIKI_W7500`, it builds without an error. but target like `TB_SENSE_1 ` fails with a proper error - bootloader can't be enabled due to configuration error.

Reply:
IF we can test this commit, we should get reduce list of  targets that are not supporting bootloader properly.

Reply:
@madchutney please review